src/gui/guiAnimatedImage.cpp

```diff
@@ -0,0 +1,83 @@
+#include "guiAnimatedImage.h"
+
+#include "client/guiscalingfilter.h"
+#include "client/tile.h" // ITextureSource
+#include "log.h"
+#include "porting.h"
+#include <string>
+
+GUIAnimatedImage::GUIAnimatedImage(gui::IGUIEnvironment *env, gui::IGUIElement *parent,
+		s32 id, const core::rect<s32> &rectangle, const std::string &name,
+		ISimpleTextureSource *tsrc) :
+		gui::IGUIElement(gui::EGUIET_ELEMENT, env, parent, id, rectangle),
+		m_name(name), m_tsrc(tsrc), m_texture(nullptr), m_global_time(0),
+		m_frame_idx(0), m_frame_count(1), m_frame_duration(1), m_frame_time(0)
+{
+	// Expected format: "texture_name:frame_count,frame_duration"
+	// If this format is not met, the string will be loaded as a normal texture
+
+	m_frame_idx = 0;
+
+	std::string::size_type colon_position = name.find(':', 0);
+	std::string::size_type comma_position = name.find(',', 0);
+
+	if (comma_position > colon_position && comma_position < name.size()) {
```

Comment:
Why isn't colon_position checked?

```suggestion
	if (comma_position != std::string::npos && colon_position != std::string::npos &&
			comma_position > colon_position) {
```

Reply:
It is...just in a dangerously indirect fashion. Good catch.